x/hard/keeper/borrow_test.go

```diff
@@ -39,389 +38,580 @@ func (suite *KeeperTestSuite) TestBorrow() {
+					previousBorrowCoins:       sdk.NewCoins(),
+					borrowCoins:               sdk.NewCoins(sdk.NewCoin("ukava", sdkmath.NewInt(20*KAVA_CF))),
+					expectedAccountBalance:    sdk.NewCoins(sdk.NewCoin("ukava", sdkmath.NewInt(20*KAVA_CF)), sdk.NewCoin("btcb", sdkmath.NewInt(100*BTCB_CF)), sdk.NewCoin("bnb", sdkmath.NewInt(100*BNB_CF)), sdk.NewCoin("xyz", sdkmath.NewInt(1))),
+					expectedModAccountBalance: sdk.NewCoins(sdk.NewCoin("ukava", sdkmath.NewInt(1080*KAVA_CF)), sdk.NewCoin("usdx", sdkmath.NewInt(200*USDX_CF)), sdk.NewCoin("busd", sdkmath.NewInt(100*BUSD_CF))),
+				},
+				{
+					borrower:                  sdk.AccAddress(crypto.AddressHash([]byte("test"))),
+					depositCoins:              []sdk.Coin{sdk.NewCoin("ukava", sdkmath.NewInt(10*KAVA_CF))},
+					borrowCoins:               sdk.NewCoins(sdk.NewCoin("ukava", sdkmath.NewInt(20*KAVA_CF))),
+					expectedAccountBalance:    sdk.NewCoins(sdk.NewCoin("ukava", sdkmath.NewInt(40*KAVA_CF)), sdk.NewCoin("btcb", sdkmath.NewInt(100*BTCB_CF)), sdk.NewCoin("bnb", sdkmath.NewInt(100*BNB_CF)), sdk.NewCoin("xyz", sdkmath.NewInt(1))),
+					expectedModAccountBalance: sdk.NewCoins(sdk.NewCoin("ukava", sdkmath.NewInt(1060*KAVA_CF)), sdk.NewCoin("usdx", sdkmath.NewInt(200*USDX_CF)), sdk.NewCoin("busd", sdkmath.NewInt(100*BUSD_CF))),
+				},
+			},
+			errArgs: []errArgs{
+				{
+					expectPass: true,
+				},
+				{
+					expectPass: true,
+				},
+			},
+		},
+		{
+			name: "first valid borrow, second insufficient funds blocks",
+			args: []args{
+				{
+					usdxBorrowLimit:           sdk.MustNewDecFromStr("100000000000"),
+					priceKAVA:                 sdk.MustNewDecFromStr("5.00"),
+					loanToValueKAVA:           sdk.MustNewDecFromStr("0.6"),
+					priceBTCB:                 sdk.MustNewDecFromStr("0.00"),
+					loanToValueBTCB:           sdk.MustNewDecFromStr("0.01"),
+					priceBNB:                  sdk.MustNewDecFromStr("0.00"),
+					loanToValueBNB:            sdk.MustNewDecFromStr("0.01"),
+					borrower:                  sdk.AccAddress(crypto.AddressHash([]byte("test"))),
+					depositCoins:              []sdk.Coin{sdk.NewCoin("ukava", sdkmath.NewInt(100*KAVA_CF))},
+					previousBorrowCoins:       sdk.NewCoins(),
+					borrowCoins:               sdk.NewCoins(sdk.NewCoin("ukava", sdkmath.NewInt(20*KAVA_CF))),
+					expectedAccountBalance:    sdk.NewCoins(sdk.NewCoin("ukava", sdkmath.NewInt(20*KAVA_CF)), sdk.NewCoin("btcb", sdkmath.NewInt(100*BTCB_CF)), sdk.NewCoin("bnb", sdkmath.NewInt(100*BNB_CF)), sdk.NewCoin("xyz", sdkmath.NewInt(1))),
+					expectedModAccountBalance: sdk.NewCoins(sdk.NewCoin("ukava", sdkmath.NewInt(1080*KAVA_CF)), sdk.NewCoin("usdx", sdkmath.NewInt(200*USDX_CF)), sdk.NewCoin("busd", sdkmath.NewInt(100*BUSD_CF))),
+				},
+				{
+					borrower:                  sdk.AccAddress(crypto.AddressHash([]byte("test"))),
+					depositCoins:              []sdk.Coin{sdk.NewCoin("ukava", sdkmath.NewInt(100*KAVA_CF))}, // too much
```

Comment:
Use of `depositCoins` and non-borrow fields after the first `args` in the list don't seem to be used in the test, so this test case seems to be not actually testing as intended

The refactor of split up args would make it easier to avoid use of fields that won't be used and make it more clear what is being tested